ligo/test/Ligo/BaseDAO/Types.hs

```diff
@@ -58,10 +64,12 @@ data ProposalL = ProposalL
 
 type CallCustomParam = (MText, ByteString)
 
-data Parameter
+-- NOTE: Constructors of the parameter types should remain sorted for the
+--'ligoLayout' custom derivation to work.
+--
```

Comment:
This is actually unnecessary, because `ligoLayout` is able of sorting the constructors as well (just as LIGO does).

These were in order only because they were written before we had `ligoLayout` and have stayed alphabetically sorted since.

Reply:
@pasqu4le I actually came across this when the build started failing for me during the derivation of 

```
data MigratableParam
  = XtzForbidden ForbidXTZParam
  | CallCustom CallCustomParam
  deriving stock (Show)
```

With the error `• Couldn't match type ‘ForbidXTZParam’ with ‘(MText, ByteString)’`. I am not sure if I am not doing anything else wrong and it still fails if the order of the constructors is as shown above.

This was fixed when I just tried re-ordering the constructors. I set out to create an issue in morley, but came across the PR for `ligolayout` that mentioned something about sorting, which gave me the impression that it was known. Re-reading it, I have no idea what gave me that impression.

[1] https://gitlab.com/morley-framework/morley/-/merge_requests/731

Reply:
@sras you are correct, it seems there is a bug in `ligoLayout` for sum-types, I have reduced it to a simple example and opened an issue in `morley` to fix it.

Please keep the comment up, but add a TODO to update it in the future mentioning [morley #527](https://gitlab.com/morley-framework/morley/-/issues/527)

Reply:
Will do.